lib/mixlib/shellout/windows/core_ext.rb

```diff
@@ -171,10 +179,19 @@ def create3(args)
       end
 
       env = nil
+      env_list = []
+
+      # Retrieve the environment variables for the specified user.
+      if hash["with_logon"]
+        logon, passwd, domain = format_creds_from_hash(hash)
+        logon_type = hash["elevated"] ? LOGON32_LOGON_BATCH : LOGON32_LOGON_INTERACTIVE
+        token = logon_user(logon, domain, passwd, logon_type)
+        env_list = retrieve_environment_variables(token, env_list)
+      end
 
       # The env string should be passed as a string of ';' separated paths.
       if hash["environment"]
-        env = hash["environment"]
+        env =  env_list.any? ? env_list : hash["environment"]
```

Comment:
It looks like `hash["environment"]` here is a string with `;` separated paths while env_list is some kind of an array. This looks like it has probably broken the old behavior.  It also needs to be merging the two of them and not overriding.  If the user has specified an environment variable in `hash["environment"]` that needs to override the setting in the env_list.  I'm not sure exactly how you'd do that, but converting both of them to hashes with key => value pairs, using Hash#merge and then rendering the Hash back into an array would probably work.